### PR TITLE
ET-4575 validate properties

### PR DESCRIPTION
**Pull request by janpetschexain**
**Reference**

- [ET-4575]

**Summary**

- update spec with property limitations
- add property validation

[ET-4575]: https://xainag.atlassian.net/browse/ET-4575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: web-api/openapi/schemas/document.yml [hunk 1]
```diff
@@ -3,8 +3,31 @@ DocumentId:
...
+  minLength: 1
+  maxLength: 256
+
 DocumentProperty:
   description: Arbitrary property that can be attached to a document.
+  oneOf:
+    - type: boolean
+    - type: number
+    - type:
+      $ref: '#/DocumentPropertyString'
```
[the remark this answers is not in the document]
Comment (janpetschexain) [reply web-api/openapi/schemas/document.yml:26]: i checked with the generated output from redocly. it seems like both approaches produce the same result. i'll go with the suggestion though, as that is what we actually want to express.

File: web-api/src/models.rs [hunk 2]
```diff
@@ -123,10 +136,75 @@ string_wrapper! {
...
 #[serde(transparent)]
-pub(crate) struct DocumentProperty(serde_json::Value);
+pub(crate) struct DocumentProperty(Value);
+
+impl DocumentProperty {
+    fn new(mut property: Value) -> Result<Self, InvalidDocumentProperty> {
+        match &mut property {
+            Value::Bool(_) | Value::Number(_) => {}
+            Value::String(string) => {
+                trim(string);
```
[the remark this answers is not in the document]
Comment (janpetschexain) [reply web-api/src/models.rs:142]: i double-checked this topic with @acrrd yesterday.

**Comment by janpetschexain** [comment]
i also moved the `check_size()` thingy into the constructor such that it can't be forgotten while you were doing the review, pls have a look.

**Review by rustonaut — APPROVED** [review]
lgtm

**Pull request merged**